### DISCUSSION TITLE
Version that supports React 17 and React 18?

### DISCUSSION THREAD

**Discussion by nrutman** [lookfirst/mui-rff#1077] [Q&A]
I'm in the process of upgrading a web application from React 17 to React 18. It seems like mui-rff version 6 makes a hard cutover to React 18.

Is it possible to get a version that supports both so we don't have to couple upgrading mui-rff with updating React itself? Is there anything in the 6.x lineage that won't work with React 17?

**Comment by lookfirst** [comment]
Probably not. Sadly, the way that the JS ecosystem works is that you have to do mass upgrades of all of this crap and it isn't pretty. For my own stuff, I tend to try to stay on the bleeding edge all the time, which is a pain, but typically less of a pain than major upgrades. At least it is a slower pain. Sorry to not have a better answer for you.
